channeldb/payments.go

```diff
@@ -67,31 +67,31 @@ var (
 
 	// paymentSequenceKey is a key used in the payment's sub-bucket to
 	// store the sequence number of the payment.
-	paymentSequenceKey = []byte("payment-sequence-key")
+	paymentSequenceKey = []byte("s") // "payment-sequence-key"
```

Comment:
I like the idea of shortening the keys. Though perhaps we should always use two characters of which at least one is not in the hex alphabet to make it a bit more clear when just looking at the keys?
For example:
```
ci -> payment-creation-info
ai -> htlc-attempt-info
si -> htlc-settle-info
fi -> htlc-fail-info
pf -> payment-fail-info

```

Reply:
Yeah, maybe two char names are better for readability. I wonder how much we'd spare by shortening our keys, since I think bbolt also serializes the key and is not dictionary based. Maybe I'm wrong though.

Reply:
Isn't all this a breaking change? Also IIRC, for etcd we hashed these keys down anyway.

Reply:
Yes, it is breaking change, but there's some huge gains even in `etcd`. With the `etcd` driver we currently hash the bucket path and then append the key and the postfix of whether this is a bucket or value key.

So something like this: `hash(hash(parent + b) + child + b) + key + b/v`. The last part is where these longs names that I renamed would be. For each payment times each HTLC.

Reply:
Another thing is that if we flatten the payment's htlc bucket that is also a change that needs migration, so we could be at least efficient breaking our schema :)

About flattening: to decide whether or not a payment is settled, we now iterate the htlc bucket (mutliple times during payment lifecycle actually) and then for each HTLC we also get the attempt, settle and fail info keys.

Which means without flattenting and assuming the above hashed structure we'd need num(htlcs) * (3 + 1) roundtrips to the DB per payment every time we need to calculate the payment status. The plus 1 comes from getting into the attempt bucket for the HTLC where the key is a global counter, so we have no way to know which ones to prefetch (keys are guaranteed to be monotonic but not sequential).